knowledge-base/pdf-from-images-with-fixedcontenteditor.md

```diff
@@ -0,0 +1,91 @@
+---
+title: How to Generate a PDF Document from Images with FixedContentEditor
+description: This article explains how to create a PDF document from a collection of images using the RadPdfProcessing library.
+type: how-to
+page_title: How to Generate a PDF Document from Images with FixedContentEditor
+slug: pdf-from-images-with-fixedcontenteditor
+tags: pdf,images,editor
+res_type: kb
+---
+# Environment
+| Version | Product | Author | 
+| --- | --- | ---- | 
+| 2024.1.124 | RadPdfProcessing|[Desislava Yordanova](https://www.telerik.com/blogs/author/desislava-yordanova)| 
+
+# Description
+This tutorial demonstrates a sample approach how to generate a PDF document from a collection of images located in a local folder.
+
+![Folder with images](images/images-folder.png)
+
+# Solution
+To create the PDF document, we will use a [FixedContentEditor]({%slug radpdfprocessing-editing-fixedcontenteditor%}) which is always associated with a single [RadFixedPage]({%slug radpdfprocessing-model-radfixedpage%}). The editor maintains an internal [Position]({%slug radpdfprocessing-concepts-position%}) inside the page at which the image block element will be inserted and drawn. The Position is adjusted after the image is rendered. If there is no remaining space on the page to draw the next image, a new page will be created and the editor's position will be moved to the beginning of the new page. 
+ 
+```csharp
+        private static void GeneratePdfFromImagesWithFixedContentEditor(string imageFolderPath)
+        {
+            Size _pageSize = new Size(Unit.MmToDip(210), Telerik.Windows.Documents.Media.Unit.MmToDip(297));
+            Padding _pageMargins = new Telerik.Windows.Documents.Primitives.Padding(
+                Unit.MmToDip(20),//left
+                 Unit.MmToDip(0),//top
+                  Unit.MmToDip(0),//right
+                   Unit.MmToDip(0));//bottom
+            RadFixedDocument fixedDocument = new RadFixedDocument();
+            RadFixedPage fixedPage = fixedDocument.Pages.AddPage();
+            fixedPage.Size = _pageSize;
+            FixedContentEditor fixedEditor = new FixedContentEditor(fixedPage);
+            double currentYposition = _pageMargins.Top;
+            double pageYLimit = _pageSize.Height - _pageMargins.Bottom;
+            string[] imageFiles = Directory.GetFiles(imageFolderPath);
+
+            foreach (string imageFilePath in imageFiles)
+            {
+                Block imageBlock = new Block();
+                imageBlock.SpacingAfter = 0;
```

Comment:
Is setting the SpacingAfter to 0 required?

Reply:
Not obligatory, but it intended to show that there is a way to control the distance between the images so they may look more distant.